### PR TITLE
Auto-reload TLS certs for druid endpoints

### LINKED ISSUE #12064: Auto-reload tls certs for druid endpoints
More often than not, the certs are refreshed quite frequently. The refresh also takes care of java store files. Druid has capability to work with these files. I would like to know how druid addresses this change at the moment. From what I see in the source code and test/integration cases, druid doesn't support it yet [please correct me if I am missing something]. 

Jetty has `KeyStoreScanner` which is capable of reloading the certs based on changes in cert files. This implementation could be very useful here. 

I have done some preliminary implementation but before I publish the PR [struggling with test cases for this particular feature], I would like to know what community thinks about this idea.

### PR DESCRIPTION
**Pull request by santosh-d3vpl3x**
Fixes #12064.

### Description

This PR has:
- [x] been self-reviewed.
   - [x] using the [concurrency checklist](https://github.com/apache/druid/blob/master/dev/code-review/concurrency.md) (Remove this item if the PR doesn't have any relation to concurrency.)
- [x] added documentation for new or modified features or behaviors.
- [x] added Javadocs for most classes and all non-trivial methods. Linked related entities via Javadoc links.
- [x] added or updated version, license, or notice information in [licenses.yaml](https://github.com/apache/druid/blob/master/dev/license.md)
- [x] added comments explaining the "why" and the intent of the code wherever would not be obvious for an unfamiliar reader.
- [x] added unit tests or modified existing tests to cover new code paths, ensuring the threshold for [code coverage](https://github.com/apache/druid/blob/master/dev/code-review/code-coverage.md) is met.

### CODE REVIEW

**Comment by santosh-d3vpl3x** [comment]
The failure is unrelated to the PR.

**Comment by santosh-d3vpl3x** [comment]
@FrankChen021 would you be interested in reviewing this PR?

File: core/src/test/java/org/apache/druid/java/util/http/client/FriendlyServersTest.java [hunk 1]
```diff
@@ -275,6 +276,8 @@ public void testFriendlySelfSignedHttpsServer() throws Exception
 
     sslConnector.setPort(0);
     server.setConnectors(new Connector[]{sslConnector});
+    KeyStoreScanner keyStoreScanner = new KeyStoreScanner(sslContextFactory);
```
Comment (a2l007) [inline core/src/test/java/org/apache/druid/java/util/http/client/FriendlyServersTest.java:279]: This wouldn't detect keystore changes if the keystore path was a soft link, would it?

**Review by a2l007 — APPROVED** [review]
@santosh-d3vpl3x Thank you for your contribution!

File: core/src/test/java/org/apache/druid/java/util/http/client/FriendlyServersTest.java (hunk 1 above)
Comment (santosh-d3vpl3x) [reply core/src/test/java/org/apache/druid/java/util/http/client/FriendlyServersTest.java:279]: Good question, https://github.com/eclipse/jetty.project/blob/jetty-10.0.x/tests/test-integration/src/test/java/org/eclipse/jetty/test/KeyStoreScannerTest.java#L182-L206 leads me to believe that it has that ability.

File: docs/operations/tls-support.md [hunk 2]
```diff
@@ -50,6 +50,8 @@ values for the configs below, among others provided by Java implementation.
 |`druid.server.https.keyStoreType`|The type of the key store.|none|yes|
 |`druid.server.https.certAlias`|Alias of TLS/SSL certificate for the connector.|none|yes|
 |`druid.server.https.keyStorePassword`|The [Password Provider](../operations/password-provider.md) or String password for the Key Store.|none|yes|
+|`druid.server.https.reloadSslContext`| Should druid server detect Key Store file change and reload.|false|no|
```
Comment (FrankChen021) [inline docs/operations/tls-support.md:53]:
```suggestion
|`druid.server.https.reloadSslContext`| Should Druid server detect Key Store file change and reload.|false|no|
```

File: docs/operations/tls-support.md [hunk 3]
```diff
@@ -50,6 +50,8 @@ values for the configs below, among others provided by Java implementation.
...
+|`druid.server.https.reloadSslContextSeconds`| how frequently should druid server scan for Key Store file change.|60|yes|
```
Comment (FrankChen021) [inline docs/operations/tls-support.md:54]:
```suggestion
|`druid.server.https.reloadSslContextSeconds`| How frequently should druid server scan for Key Store file change.|60|yes|
```

**Review by FrankChen021 — APPROVED** [review]

**Comment by FrankChen021** [comment]
Thank you @santosh-d3vpl3x for your first contribution.

**Comment by gianm** [comment]
Going to close and reopen this to try to retrigger a Travis CI run. It's using an older version of `.travis.yml` that refers to a no-longer-existing `it-tools` project, & so can't build.

**Comment by gianm** [comment]
@santosh-d3vpl3x Could you merge master into your branch, please? I think for Travis CI to run successfully the branch needs to pull in a `.travis.yml` update from #12368. Thank you.

**Comment by santosh-d3vpl3x** [comment]
> @santosh-d3vpl3x Could you merge master into your branch, please? I think for Travis CI to run successfully the branch needs to pull in a `.travis.yml` update from #12368. Thank you.

It looks stuck!

**Comment by gianm** [comment]
> It looks stuck!

There's a queue sometimes, since we share one Travis CI account across all Apache projects. Looks like the job is running now!

**Pull request merged**